### ISSUE TITLE
getDB in mongo-java-driver 3

### ISSUE THREAD

**Issue by kgignatyev** [devbliss/mongojack#105]
It looks like getDB() call is deprecated and its replacement does not seem to provide any way to get DBCollection which is effectively get deprecated too
https://jira.mongodb.org/browse/JAVA-1331

http://mongodb.github.io/mongo-java-driver/3.0/driver/reference/crud/

Are there any plans in place to migrate to use java-driver-3 API?

**Comment by benmccann** [comment]
See https://github.com/devbliss/mongojack/pull/104

**Comment by kgignatyev** [comment]
Not quite, mogojack works with 3.0.0 driver fine provided that appropriate exclusions set in maven project. My concern is the entire API that mongojack seems to utilize is deprecated in 3.0.0 driver

**Comment by tmcdonnell87** [comment]
+1

**Comment by spiritoosSanki** [comment]
+1 !

**Comment by benmccann** [comment]
There are 0 deprecation warnings in mongojack: https://github.com/devbliss/mongojack/commit/395f69511e490a64bfb1343f34ebe4e269d81ce7

However, maybe the lack of deprecation warnings is a result of a mistake in the driver? https://jira.mongodb.org/browse/JAVA-1858

**Comment by tmcdonnell87** [comment]
You're correct. There may not be deprecation warnings in mongojack itself, but in order to initialize the library, I have to get the databse using the deprecated getDB method. 

		mongoClient = new MongoClient( "localhost" , 27017 );
		//database = mongoClient.getDatabase("ecomm");  3.x and beyond
		database=mongoClient.getDB("ecomm");   //deprecated

**Comment by benmccann** [comment]
Yeah, sent a PR to make it clearer that DB is deprecated: https://github.com/mongodb/mongo-java-driver/pull/318

**Comment by BenRomberg** [comment]
@benmccann: What's the upgrade strategy for mongojack? Is the next version still going to be compatible to mongo-java-driver 2.x or are preparations already ongoing to move to the new mongo-java-driver 3.x API?

I'd love to use the new API, it looks a lot more typesafe, with special types for all the query options. Can also help with the move to the new API if desired.

**Comment by benmccann** [comment]
It's already upgraded to use 3.x in master. Just haven't released the latest code yet

**Comment by olmaga** [comment]
When is a release of 3.x-compatibility planned? (Heroku sandboxes stop working with 2.6 next weeks...)

**Comment by benmccann** [comment]
@olmaga just cut a release. if you use mongojack 2.5 it should work

**Comment by olmaga** [comment]
@benmccann 2.5 does not look that stable - even the test cases use the deprecated code. (e.g. mongo.getDB(...))
or is even the deprecated code working on 3.0?

**Comment by benmccann** [comment]
Deprecated code still works. References to deprecated code can be removed in future releases

**Comment by mancvso** [comment]
Doesn't seem to work with the async driver either.
